assignment3/bookstore.js

```diff
@@ -0,0 +1,145 @@
+import BookInfo from './BookInfo.js';
+
+const bookList = document.querySelector('.section_ul');
+const categoryList = document.querySelector('.category_list');
+
+const checkedCategory = [];
+const checkboxes = document.querySelectorAll('.check_category');
+const checkAll = document.querySelector('.all');
+const checkWeb = document.querySelector('.web');
+const checkSelf = document.querySelector('.self');
+const checkNovel = document.querySelector('.novel');
+
+let filterBook = [];
+
+//filterBook을 렌더링하는 함수
+const renderBooks = () => {
+    filterBook.forEach((book) => {
+        const bookItem = document.createElement('li');
+        bookItem.classList.add('section_book', 'pulse');
+        bookItem.innerHTML = `
+    <header class="section_book_header"><h2>${book.title}</h2></header>
+    <div class="modal"><p class="modalHash">${book.hashtag}</p> <button class="closeBtn"><img src="images/x-button.png"  width="10" height="10"alt="닫기버튼"></img></button></div>
+    <span class="hashtag_wrapper">
+    <p class="hideHash">${book.hashtag}</p>
+    <button type="button" class="hashPlusButton"><img src="images/plus_button.png" width="25" height="25" alt="플러스버튼"></img></button>
+    </span>
+    <img class="book_img"src="${book.image}" alt="${book.title}이미지">
+    <i class="fas fa-solid fa-heart"></i>
+  `;
+
+        const hashPlusButton = bookItem.querySelector('.hashPlusButton');
+        const modal = bookItem.querySelector('.modal');
+        const closeBtn = bookItem.querySelector('.closeBtn');
+
+        hashPlusButton.addEventListener('click', (e) => {
+            e.preventDefault();
+            modal.style.display = 'flex';
+        });
+
+        // 모달 닫기 버튼에 클릭 이벤트 추가
+        closeBtn.addEventListener('click', () => {
+            // 모달 숨기기
+            modal.style.display = 'none';
+        });
+
+        bookList.appendChild(bookItem);
+    });
+};
+
+//상단 카테고리 리스트를 삭제하는 함수
+const removeCategory = (categoryName, nav_category) => {
+    const p = nav_category.querySelector('p');
+    if (p.textContent === categoryName) {
+        nav_category.remove();
+        const index = checkedCategory.indexOf(categoryName);
+        if (index !== -1) {
+            checkedCategory.splice(index, 1);
+        }
+        filterBooks(checkedCategory);
+    }
+
+    if (categoryName === '전체') {
+        checkAll.checked = false;
+    } else if (categoryName === '웹개발서적') {
+        checkWeb.checked = false;
+    } else if (categoryName === '자기개발서적') {
+        checkSelf.checked = false;
+    } else if (categoryName === '소설') {
+        checkNovel.checked = false;
+    }
```

Comment:
```suggestion
switch (categoryName) {
  case '전체':
    checkAll.checked = false;
    break;
  case '웹개발서적':
    checkWeb.checked = false;
    break;
  case '자기개발서적':
    checkSelf.checked = false;
    break;
  case '소설':
    checkNovel.checked = false;
    break;
  default:
    break;
}
```
요건 진짜 개인 성향 차이일 수도 있는데!
저는 단순히 값의 일치 여부를 판단하는 용도에서는 switch-case를 쓰는게 조금 더 가독성 있게 느껴지더라구요! ㅎㅎ
조건의 경우도 여러 개이기 때문에 switch-case도 한번 고려해봐도 좋을거같아용!!

Reply:
맞아요! switch문도 한 번 고려해 보는 게 좋을 것 같아요!

Reply:
흠 맞아요 저도 switch case가 좀 더 가독성이 좋을 수도 있다고 생각합니다! 다음엔 그런 가독성 부분도 고려하면서 코드를 작성해보겠습니다 :)

Reply:
케이스에대한 상수파일작성도 때로는 갠찬을수이따아